### PR TITLE
Documentation Update pulled from Refactor #2

### PR DESCRIPTION
**Pull request by elehmer**
# Description of PR

Documentation updates pulled from refactor phase 2.

**Summary of changes and related issue**

Fleshes out the core and util modules with more documentation.

**Relevant motivation and context**

**Dependencies required for this change?**

**Fixes # (issue), delete if not necessary**

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ x ] This change requires a documentation update ;)

# How Has This Been Tested?

**Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.**

Not tested...

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ x ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by nicolejkeeney — APPROVED** [review]
This is such a huge improvement to the documentation, thanks for all the effort Eric!

File: climakitae/core/data_load.py [hunk 1]
```diff
@@ -1093,7 +1070,7 @@ def read_catalog_from_csv(selections, csv, merge=True):
         Default to True.
 
     Returns
-    ----------
+    -------
     one of the following, depending on csv input and merge:
```
Comment (Tianchi-Liu) [inline climakitae/core/data_load.py:1074]:
```suggestion
```

File: climakitae/core/data_load.py [hunk 2]
```diff
@@ -1093,7 +1070,7 @@ def read_catalog_from_csv(selections, csv, merge=True):
...
```
Comment (Tianchi-Liu) [inline climakitae/core/data_load.py:1075]:
```suggestion
```

File: climakitae/core/data_export.py [hunk 3]
```diff
@@ -705,6 +728,15 @@ def _metadata_to_file(ds, output_name):
     """
     Write NetCDF metadata to a txt file so users can still access it
     after exporting to a CSV.
+
+    Parameters
+    ----------
+    ds: xr.DataSet
```
Comment (Tianchi-Liu) [inline climakitae/core/data_export.py:734]:
```suggestion
    ds: xr.DataArray or xr.Dataset
```

File: climakitae/core/data_interface.py [hunk 4]
```diff
@@ -1592,7 +1759,17 @@ def _display_select(self):
...
+    """Get a user-friendly version of the intake data catalog using climakitae naming conventions
+
+    Parameters
+    ----------
+    intake_catalog: intake_esm.source.ESMDataSource
+    variable_descriptions: pd.DataFrame
+
+    Returns
+    -------
+    cat_df_cleaned: intake_esm.source.ESMDataSource
```
Comment (Tianchi-Liu) [inline climakitae/core/data_interface.py:1771]:
```suggestion
    cat_df_cleaned: pd.DataFrame
```

File: climakitae/core/data_interface.py [hunk 5]
```diff
@@ -1749,25 +1924,24 @@ def get_data_options(
 
     Parameters
     ----------
-    variable: str, optional
+    variable: str (optional)
```
Comment (Tianchi-Liu) [inline climakitae/core/data_interface.py:1927]: `, optional` is styled better. Something worth cleaning up probably only when convenient.

**Review by Tianchi-Liu — APPROVED** [review]
Confirming that all your hard work will show up on the doc!

An observation is that the param objects in `DataParameters` are documented twice. They autogenerate their own “documentation” containing the default parameter values ([example](https://climakitae.readthedocs.io/en/latest/climakitae.core.html#climakitae.core.data_interface.DataParameters)). I don’t know if they can be removed, but can look into it.

Also double documented are the public methods of `Boundaries`. Sphinx autodocuments them, so method docstrings are sufficient. No need to add them to class docstrings.

For `Boundaries`, private methods are also documented, which is against the style guide, but the methods make sense for me to be made public.

A final thought- the module docstrings and the info in the ascii art (in `data_load.py`) that are deleted can be helpful, especially for new contributors.

Minor tweaks below. See what you think!

**Comment by elehmer** [comment]
@Tianchi-Liu Would you be willing to take the lead on making the changes you think are good for the final rendering in readthedocs? If so, feel free to make commits to this branch.

**Comment by Tianchi-Liu** [comment]
> @Tianchi-Liu Would you be willing to take the lead on making the changes you think are good for the final rendering in readthedocs? If so, feel free to make commits to this branch.

@elehmer I fixed the duplicate issues I brought up for `DataParameters` and `Boundaries`. I also applied some small code suggestions and marked them as resolved. Here’s a [preview of readthedocs](https://climakitae--404.org.readthedocs.build/en/404/climakitae.core.html#). Let me know of additional changes needed!

**Comment by elehmer** [comment]
> > @Tianchi-Liu Would you be willing to take the lead on making the changes you think are good for the final rendering in readthedocs? If so, feel free to make commits to this branch.
> 
> @elehmer I fixed the duplicate issues I brought up for `DataParameters` and `Boundaries`. I also applied some small code suggestions and marked them as resolved. Here’s a [preview of readthedocs](https://climakitae--404.org.readthedocs.build/en/404/climakitae.core.html#). Let me know of additional changes needed!

This looks great. Thank you! Let's go ahead and merge this

**Pull request merged**